src/grid/grid.component.ts

```diff
@@ -0,0 +1,630 @@
+        }
+        this.colDiffer = this.cols.changes.subscribe((_) => {
+            this.checkColumns();
+            this._detector.markForCheck();
+        });
+
+    }
+
+    /**
+     * @hidden
+     */
+    protected checkColumns(): void {
+        this.columns = this.cols.toArray();
+        this.columns.forEach((col, index) => {
+            col.index = index;
+            this.onColumnInit.emit({column: col});
+        });
+    }
+
+    /**
+     * @hidden
+     */
+    public ngDoCheck(): void {
+        let diff: IterableDiffer = this._differ.diff(this.data);
+        if (this.shouldDataBind || diff) {
+            this.onBeforeProcess.emit(this.dataContainer.state);
+            this.dataContainer.process();
+            this.shouldDataBind = false;
+        }
+    }
+
+    /**
+     * @hidden
+     */
+    public ngOnDestroy(): void {
+        this.colDiffer.unsubscribe();
+    }
+
+    public getColumnByIndex(index: number): any {
+        if (this.columns[index] !== undefined) {
+            return this.columns[index];
+        }
+    }
+
+    public getColumnByField(field: string): any {
+        return this.columns.find((col) => col.field === field);
+    }
+
+    /**
+     * Returns the cell at rowIndex/columnIndex.
+     *
+     * @param {number} rowIndex
+     * @param {number} columnIndex
+     * @returns
+     *
+     * @memberOf IgxGrid
+     */
+    public getCell(rowIndex: number, columnIndex: number): any {
+        let result: any = { dataItem: null, element: null };
+        let record: any = this.dataContainer.getRecordByIndex(rowIndex, DataAccess.TransformedData);
+        let element: any = this._elementRef.nativeElement.querySelector(`td[data-row="${rowIndex}"][data-col="${columnIndex}"]`);
+
+        if (record) {
+            result.dataItem = record[this.columns[columnIndex].field];
+        }
+
+        if (element) {
+            result.element = element;
+        }
+
+        return result;
+    }
+
+    /**
+     * Returns
+     *
+     * @param {number} rowIndex
+     * @returns
+     *
+     * @memberOf IgxGrid
+     */
+    public getRow(rowIndex: number): any {
+        let result: any = { row: null, element: null };
+        let record: any = this.dataContainer.getRecordByIndex(rowIndex, DataAccess.TransformedData);
+        let element: any = this._elementRef.nativeElement.querySelector(`tbody > tr[data-row="${rowIndex}"]`);
+
+        if (record) {
+            result.row = record;
+        }
+
+        if (element) {
+            result.element = element;
+        }
+
+        return result;
+    }
+
+    /**
+     * Focuses the grid cell at position row x column.
+     * Fires the onCellSelection event.
+     *
+     * @param {(number | string)} rowIndex      : the index of the row
+     * @param {(number | string)} columnIndex   : the index of the column
+     *
+     * @memberOf IgxGrid
+     */
+    public cellAt(rowIndex: number | string, columnIndex: number | string): void {
```

Comment:
This method focuses a cell, so it shouldn't be called `cellAt` because the naming infers that it would return a cell.